dialogue-apache-hc4-client/src/main/java/com/palantir/dialogue/hc4/ApacheHttpClientChannels.java

```diff
@@ -231,7 +164,138 @@ public void close() throws IOException {
 
         @Override
         public String toString() {
-            return "CloseableClient{client=" + client + '}';
+            return "CloseableClient{client="
+                    + client + ", leakDetector="
+                    + leakDetector + ", executor="
+                    + executor + '}';
+        }
+    }
+
+    public static ClientBuilder clientBuilder() {
+        return new ClientBuilder();
+    }
+
+    public static final class ClientBuilder {
+
+        @Nullable
+        private ClientConfiguration clientConfiguration;
+
+        @Nullable
+        private String clientName;
+
+        @Nullable
+        private ExecutorService executor;
+
+        private ClientBuilder() {}
+
+        public ClientBuilder clientConfiguration(ClientConfiguration value) {
+            this.clientConfiguration = Preconditions.checkNotNull(value, "ClientConfiguration is required");
+            return this;
+        }
+
+        public ClientBuilder clientName(String value) {
+            this.clientName = Preconditions.checkNotNull(value, "ClientConfiguration is required");
```

Comment:
```suggestion
            this.clientName = Preconditions.checkNotNull(value, "clientName is required");
```

Reply:
I remember also @mattsills was asking about the significance of this field. Perhaps this is also worth some javadoc to say this doesn't influence request behaviour, just used for metrics?

Reply:
oof, good catch. I'm falling apart!